README.md

```diff
@@ -3,289 +3,146 @@
+When developing a web application, defining some configuration is recommended. Simple applications may work with the CLI's default settings, but following the [Application Shell Architecture](https://developers.google.com/web/updates/2015/11/app-shell) for your app and manually defining an application shell and fragments will give you the best performance in production.
 
-There are two type of projects:
+Here's a brief summary of the configuration options you can use to describe your web application structure:
 
-* Elements projects
+  - [`entrypoint`](https://www.polymer-project.org/2.0/docs/tools/polymer-json#entrypoint) (Defaults to `index.html`): The main entrypoint to your app.
+  - [`shell`](https://www.polymer-project.org/2.0/docs/tools/polymer-json#shell) (Optional): The app shell.
+  - [`fragments`](https://www.polymer-project.org/2.0/docs/tools/polymer-json#fragments) (Optional): A list of other entrypoints into your application.
+  - `root` (Defaults to current working directory): The web root of your application, can be a subfolder of your project directory.
+  - `sources` (Defaults to `src/**/*`): The source files in your application.
 
-  Element projects contain one or more reusable element definitions, intended to be used from other elements, applications or pages. Element definitions live at the top level of the project so they are easy to import. Elements import their dependencies with relative paths that reference sibling folders to the project folder.
+Configuration can be passed for all commands via the global CLI: `--entrypoint`, `--shell`, etc. However we recommend saving your configuration to a `polymer.json` configuration file in your project. This guarantees a single shared configuration that will be read automatically for every command. Other project settings, like build and lint rules, can also be defined here.
 
-* Application projects
+Read the [polymer.json spec](https://www.polymer-project.org/2.0/docs/tools/polymer-json) for a full list of all supported fields with examples.
 
-  Application projects are self-contained and intended to be deployed as a standalone website. Application projects contain elements in a `src/` folder and import their dependencies with absolute paths, or relative paths that reference folders inside the project folder.
 
-### Application Styles
+## Command Overview
 
-Polymer-CLI currently supports two styles of applications:
+### `polymer help [COMMAND]`
 
-  * Monolithic applications, which have a single entrypoint (usually index.html) and eagerly import all dependencies.
+Run `polymer help` to get a helpful list of supported commands. Pass it a command name (ex: `polymer help serve`) to get detailed information about that command and the options it supports.
 
-  * "App shell" applications, which have a very lightweight entrypoint, an app-shell with startup and routing logic, and possibly lazy loaded fragments.
 
-### App-shell Structure
+### `polymer init`
```

Comment:
This was already discussed here: https://github.com/Polymer/polymer-cli/pull/675#discussion_r112048237

I'm okay with either, but I'm going to add it back for now to match the rest of the document. Happy to discuss more and remove again if we still feel its not useful here.